mineral-ui/mineral-ui
# [Themes] Expose "withTheme" HOC

### Description

* Exposes `withTheme` higher order component via `mineral-ui/themes`
* Updates theming page to document `withTheme` API

### Motivation and context

Primarily to increase discoverability of this useful component.

closes #676

### Screenshots, videos, or demo, if appropriate

https://676-with-theme--mineral-ui.netlify.com/theming

### How to test

1. Review the `withTheme` section of the theming page.
2. Maybe try out the example code in CRA.
3. Verify that withTheme can be imported via ES and CJS

### Types of changes

- New feature (non-breaking change which adds functionality)

### Checklist
